src/deep_search.py

```diff
@@ -1,43 +1,49 @@
 import csv
-from googleapiclient.discovery import build
 from urllib.parse import urlparse
 
+from googleapiclient.discovery import build
+
+
 def _init(key):
-    cse = build(
-        "customsearch",
-        "v1",
-        developerKey = key
-    )
+    cse = build("customsearch", "v1", developerKey=key)
     return cse
 
+
 def _longest_common_path(a, b):
-    i,j = 0
+    i = j = 0
     while a[i] == b[i]:
         if i + 1 == len(a) or i + 1 == len(b):
             break
-        if a[i] == '/': j = i
+        if a[i] == "/":
```

Comment:
Not too sure what the expectation is for this function. It may not work correctly if URLs do not have a "/". 

If we run it for "abd.com", "abc.com" then it returns just a. What's the expectation here? Should it return None instead? (I could be wrong).

Reply:
When we fetch URLs from google custom search API, the URLs are ending with a "/" at all times.

Reply:
Also, the arguments of this function can only be paths either full or just "/" for no path, no domains in any case.